extensions/integration-tests/src/notebook.util.ts

```diff
@@ -72,8 +72,11 @@ export const pythonKernelMetadata = {
 };
 
 export function writeNotebookToFile(pythonNotebook: azdata.nb.INotebookContents): vscode.Uri {
+	var path = require('path');
 	let notebookContentString = JSON.stringify(pythonNotebook);
-	let localFile = tempWrite.sync(notebookContentString, 'notebook.ipynb');
+	let localFile = path.join(process.cwd(), 'notebook' + Math.floor(Math.random() * 101) + '.ipynb');
+	console.log(`Local file is created: '${localFile}'`);
+	fs.writeFileSync(localFile, notebookContentString);
```

Comment:
Not sure if it matters for this scenario but keep in mind fs sync methods hault the entire thread (and process) until it finishes. promisefying the non-sync method is the preferred approach.

Reply:
we use fs-extra all over the code for this since it has promise implementation already done.

Reply:
It's also not hard to just promise a callback style manually. https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/base/common/async.ts#L665